=== modified file 'base_calendar/base_calendar.py'
--- base_calendar/base_calendar.py	2012-03-05 14:35:54 +0000
+++ base_calendar/base_calendar.py	2014-03-24 10:11:06 +0000
@@ -947,7 +947,14 @@
             duration = 24.0
             if start_date:
                 start = datetime.strptime(start_date, "%Y-%m-%d %H:%M:%S")
-                start_date = datetime.strftime(datetime(start.year, start.month, start.day, 0,0,0), "%Y-%m-%d %H:%M:%S")
+                #start_date = datetime.strftime(datetime(start.year, start.month, start.day, 0,0,0), "%Y-%m-%d %H:%M:%S")
+                # change start_date's time to 00:00:00 in the user's timezone
+                user = self.pool.get('res.users').browse(cr, uid, uid)
+                tz = pytz.timezone(user.context_tz) if user.context_tz else pytz.utc
+                start = pytz.utc.localize(start).astimezone(tz)     # convert start in user's timezone
+                start = start.replace(hour=0, minute=0, second=0)   # change start's time to 00:00:00
+                start = start.astimezone(pytz.utc)                  # convert start back to utc
+                start_date = start.strftime("%Y-%m-%d %H:%M:%S")
                 value['date'] = start_date
 
 

