=== modified file 'board/board.py'
--- board/board.py	2012-02-08 01:27:26 +0000
+++ board/board.py	2013-05-27 10:35:05 +0000
@@ -119,7 +119,6 @@
         res = {}
         res = super(board_board, self).fields_view_get(cr, user, view_id, view_type,\
                                  context, toolbar=toolbar, submenu=submenu)
-
         vids = self.pool.get('ir.ui.view.custom').search(cr, user,\
                      [('user_id', '=', user), ('ref_id' ,'=', view_id)])
         if vids:
@@ -140,6 +139,7 @@
                     node.remove(child)
                 else:
                     child=remove_unauthorized_children(child)
+            self.trans_processing(cr, user, node, context=context)
             return node
 
         def encode(s):
@@ -149,8 +149,34 @@
 
         archnode = etree.fromstring(encode(arch))
         return etree.tostring(remove_unauthorized_children(archnode),pretty_print=True)
-
-
+    
+    def trans_processing(self, cr, user, node, context=None):
+        if context is None:
+            context = {}
+        if 'lang' in context:
+            if node.get('string') :
+                trans = self.pool.get('ir.translation')._get_source(cr, user, self._name, 'view', context['lang'], node.get('string'))
+                if trans == node.get('string') and ('base_model_name' in context):
+                    trans = self.pool.get('ir.translation')._get_source(cr, user, context['base_model_name'], 'view', context['lang'], node.get('string'))
+                if trans:
+                    node.set('string', trans)
+            if node.get('confirm'):
+                trans = self.pool.get('ir.translation')._get_source(cr, user, self._name, 'view', context['lang'], node.get('confirm'))
+                if trans:
+                    node.set('confirm', trans)
+            if node.get('sum'):
+                trans = self.pool.get('ir.translation')._get_source(cr, user, self._name, 'view', context['lang'], node.get('sum'))
+                if trans:
+                    node.set('sum', trans)
+            if node.get('avg'):
+                trans = self.pool.get('ir.translation')._get_source(cr, user, self._name, 'view', context['lang'], node.get('avg'))
+                if trans:
+                    node.set('avg', trans)
+            if node.get('help'):
+                trans = self.pool.get('ir.translation')._get_source(cr, user, self._name, 'view', context['lang'], node.get('help'))
+                if trans:
+                    node.set('help', trans)
+            return
 
 
     _columns = {

